bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/AbstractMQTTThingHandler.java

```diff
@@ -121,15 +121,21 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
 
         if (data.isReadOnly()) {
-            logger.warn("Channel {} is a read-only channel, ignoring command {}", channelUID, command);
+            logger.trace("Channel {} is a read-only channel, ignoring command {}", channelUID, command);
             return;
         }
 
         final CompletableFuture<Boolean> future = data.publishValue(command);
-        future.exceptionally(e -> {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getLocalizedMessage());
-            return false;
-        }).thenRun(() -> logger.debug("Successfully published value {} to topic {}", command, data.getCommandTopic()));
+        future.handle((v, ex) -> {
+            if (ex != null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getLocalizedMessage());
+                logger.debug("Failed publishing value {} to topic {}: {}", command, data.getCommandTopic(), ex
+                        .getMessage());
```

Comment:
Formatting looks off. Is this debug statement necessary if we already update the Thing status?

Reply:
value and topic might be valuable information in debugging. the thing status pnly reflect the error message.

Reply:
I generally would say that such information should go into the thing status if possible, so that you don't even need to consult the log.